core/app.cc

```diff
@@ -360,13 +363,25 @@ void application::prepare_argv(elf::program *program)
     }
 
     // Load vdso library if available
-    int auxv_parameters_count = 2;
+    int auxv_parameters_count = 4;
     _libvdso = program->get_library("libvdso.so");
     if (!_libvdso) {
         auxv_parameters_count--;
         WARN_ONCE("application::prepare_argv(): missing libvdso.so -> may prevent shared libraries specifically Golang ones from functioning\n");
     }
 
+    // Initialize random bytes array so it can be passed as AT_RANDOM auxv vector
+    if (getrandom(random_bytes, sizeof(random_bytes), GRND_RANDOM) != sizeof(random_bytes)) {
+        // Fall back to rand_r()
+        u64 time_now = std::chrono::duration_cast<std::chrono::nanoseconds>(
+            osv::clock::wall::now().time_since_epoch()).count();
+        for (unsigned idx = 0; idx < sizeof(random_bytes)/(sizeof(int)); idx ++) {
+            // Point to the least significant part of time time as seed
+            reinterpret_cast<int*>(random_bytes)[idx] = rand_r(reinterpret_cast<unsigned int*>(&time_now) + 1);
+            time_now++; // Make sure each time different seed is passed to rand_r()
```

Comment:
I don't understand this line. rand_r takes the seed you point it to, and also writes the next seed back to it. You don't need to increment the seed...

Reply:
Once I follow you suggestion above this become irrelevant.